test/e2e/setup/rosa/setup/maestro.sh

```
@@ -56,81 +58,213 @@ maestro_cert_arn=$(aws iot create-keys-and-certificate \
+# RDS requires at least 2 AZs for DB subnet group
+if [ ${#unique_azs[@]} -lt 2 ]; then
+    echo "DB subnet group requires at least 2 AZs. Creating additional subnet..."
+
+    # Get VPC CIDR to determine available IP ranges
+    vpc_cidr=$(aws ec2 describe-vpcs --region=${region} --vpc-ids ${vpc} --output json | jq -r '.Vpcs[0].CidrBlock')
+    echo "VPC CIDR: $vpc_cidr"
+
+    # Find an AZ that doesn't have a subnet yet
+    target_az=""
+    for az in $(aws ec2 describe-availability-zones --region=${region} --output json | jq -r '.AvailabilityZones[].ZoneName'); do
+        if [[ ! " ${unique_azs[@]} " =~ " ${az} " ]]; then
```

Shellcheck (0.11.0)

[error] Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @).

(SC2199)